src/ui/castleinternaljoystickswindows.pas

```diff
@@ -245,8 +249,18 @@ procedure TWindowsJoysticksBackend.Poll(const List: TJoystickList;
       FillChar( Joystick.State.Axis[ JOY_POVX ], 8, 0 );
       if ( Joystick.Info.Caps and JOY_HAS_POV > 0 ) and ( state.dwPOV and $FFFF <> $FFFF ) then
       begin
-        Joystick.State.Axis[ JOY_POVX ] := Round( Sin( state.dwPOV and $FFFF div 100 ) );
-        Joystick.State.Axis[ JOY_POVY ] := -Round( Cos( state.dwPOV and $FFFF div 100 ) );
+        _value := Round( Sin( state.dwPOV and $FFFF div 100 / 180 * Pi ) );
```

Comment:
- To convert from/to radians, it's best to use DegToRad / RadToDeg from Math unit. This way it looks obvious to the reader :)

- The `Round` at the end seems unnecessary (possibly a leftover of some previous refactor -- I realize it was there before your changes). Since we operate on float values, can we just remove this `Round` and have smooth values? Right now, this `Round` unnecessarily changes the value to -1 / 0 / 1, if I read the code right?

- Also when calculating argument for Sin / Cos, use `/` instead of `div` since we are operating on floating point values anyway, so `(state.dwPOV and $FFFF) / 100.0` seems better (doesn't lose any fractional information from the result of divison).

Reply:
All in all, I think this line could be

```
_value := Sin( DegToRad((state.dwPOV and $FFFF) / 100.0) );
```

Similar comment to  line with `Cos` below.
 Round( Sin( state.dwPOV and $FFFF div 100 / 180 * Pi ) );

Reply:
Well, maybe we can go further? Now notation of the database starts being clear for me: dpdown:h0.4,dpleft:h0.8,dpright:h0.2,dpup:h0.1. They simply make it from dwPOV without any sin/cos conversions - 0.1*$FFFF is dpup; 0.2*$FFFF is dpright and so on. That doesn't correspond well to the reported values (we have 0, Pi/2, Pi, 3*Pi/2 - i.e. equally spaced; and they have 0.1, 0.2, 0.4, 0.8 i.e. 2^n/10, n=0..3). But anyway we can think of it this way.

Let me try if multiple D-Pad keys can be pressed simultaneously and whether it provides a consistent result (then our current sin/cos approach is indeed better).

Reply:
Tested everything and it looks like our current approach is better and safer :) Fixed according to your suggestions.